From: "Uladzislau Rezki (Sony)" <urezki@gmail.com>
Date: Sun, 20 Apr 2025 16:20:29 +0200
Subject: [PATCH] lib/test_vmalloc.c: fix compile error with CONFIG_TINY_RCU

When the kernel is built with CONFIG_TINY_RCU, compiling the
test_vmalloc.c fails, due to conflicting definitions like "struct
srcu_usage", etc.  between "srcutiny.h" and "srcutree.h".

It happens because of direct including the "srcutree.h" bypassing the
automatic selection mechanism.  A correct approach is to include a generic
<linux/srcu.h> which properly select SRCU implementation based on a kernel
configuration.

Fix it by replacing the "srcutree.h" include with "srcu.h".

Link: https://lkml.kernel.org/r/20250420142029.103169-1-urezki@gmail.com
Signed-off-by: Uladzislau Rezki (Sony) <urezki@gmail.com>
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202504190717.tJHs381f-lkp@intel.com/
Cc: Baoquan He <bhe@redhat.com>
Cc: Christop Hellwig <hch@infradead.org>
Cc: Mateusz Guzik <mjguzik@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 lib/test_vmalloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/test_vmalloc.c b/lib/test_vmalloc.c
index 4ab23e5e772d..399751022eea 100644
--- a/lib/test_vmalloc.c
+++ b/lib/test_vmalloc.c
@@ -15,7 +15,7 @@
 #include <linux/delay.h>
 #include <linux/mm.h>
 #include <linux/rcupdate.h>
-#include <linux/srcutree.h>
+#include <linux/srcu.h>
 #include <linux/slab.h>
 
 #define __param(type, name, init, msg)		\
-- 
2.39.5

